src/components/scenes/TransactionDetailsScene.tsx

```diff
@@ -31,62 +23,42 @@ import { Airship, showError, showToast } from '../services/AirshipInstance'
 import { cacheStyles, Theme, ThemeProps, useTheme } from '../services/ThemeContext'
 import { EdgeText } from '../themed/EdgeText'
 import { MainButton } from '../themed/MainButton'
-import { SwapDetailsTiles } from '../themed/SwapDetailsTiles'
+import { SwapDetailsTiles } from '../tiles/SwapDetailsTiles'
 import { Tile } from '../tiles/Tile'
+import { TransactionCryptoAmountTile } from '../tiles/TransactionCryptoAmountTile'
+import { TransactionFiatTiles } from '../tiles/TransactionFiatTiles'
 
 interface OwnProps {
   navigation: NavigationProp<'transactionDetails'>
   route: RouteProp<'transactionDetails'>
   wallet: EdgeCurrencyWallet
 }
 interface StateProps {
-  contacts: GuiContact[]
-  currentFiatAmount: string
-  subcategoriesList: string[]
-  walletDefaultDenomProps: EdgeDenomination
+  thumbnailPath?: string
 }
 interface DispatchProps {
-  getSubcategories: () => void
   refreshTransaction: (walletId: string, transaction: EdgeTransaction) => void
-  setNewSubcategory: (newSubcategory: string) => void
 }
 type Props = OwnProps & StateProps & DispatchProps & ThemeProps
 
 interface State {
   acceleratedTx: EdgeTransaction | null
   direction: string
-  thumbnailPath?: string
 
   // EdgeMetadata:
-  amountFiat: number
   bizId: number
   category: string
   name: string
   notes: string
 }
 
-interface FiatCryptoAmountUI {
-  amountString: string
-  symbolString: string
-  feeString: string
-}
-
-interface FiatCurrentAmountUI {
-  amount: string
-  difference: string
-  percentage: string
-}
-
-const getAbsoluteAmount = (edgeTransaction: EdgeTransaction): string =>
-  edgeTransaction && edgeTransaction.nativeAmount ? abs(edgeTransaction.nativeAmount) : ''
-
 // Only exported for unit-testing purposes
 class TransactionDetailsComponent extends React.Component<Props, State> {
   constructor(props: Props) {
     super(props)
-    const { amountFiat: defaultAmountFiat = 0, edgeTransaction, thumbnailPath } = props.route.params
+    const { edgeTransaction } = props.route.params
```

Comment:
Isn't the `defaultAmountFiat` route param used by the send scene? I recall there is some weird logic to how a new trasnsaction's amount metadata propagates from the send scene to the details scene. If you tested with a send and everything looked good, then there is no concern here.

Reply:
The transaction list is the only place in the app who passes this param. I am guessing this was done because the transaction list already has to look up the historic rate anyhow, so it was easier to pass it in than duplicate the logic.

Now that we have hooks, we *can* duplicate the logic, and we can make it live-updating instead of relying on the one-shot `componentDidMount`. I don't fully understand why the old code was failing, since all the dots were being connected, but some sort of race condition was preventing the data from getting where it needed to go.

Reply:
Okay, that's a great explanation. I really tried to dig to figure out what the bug was, but if it's one of those obscure issue that's resolved by just simply cleaning up the code, then that's well and good. This PR has solid changes that all look right to me, the only concern is with the fragility around this code and possible regressions there. But this isn't a large concern because I feel like these issues will be resolved during QA and we should be pushing this change out and making things more robust anyway.

All of it is approved with no changes that I can perceive to be needed.